### PR TITLE
[Rust Frontend] Validate tokenized bad_words vocabulary range

### PR DESCRIPTION
**Pull request by reidliu41**
Rust sampling parameter lowering.

  This matches Python behavior and rejects out-of-vocab bad word token IDs
  before they are forwarded to engine-core.

## Purpose

  Fix a Rust frontend validation gap for `bad_words`.

The Rust lowering path tokenizes `bad_words` into `_bad_words_token_ids`, but previously did not validate those token IDs before forwarding sampling params to engine-core. Python validates the tokenized bad word IDs against the tokenizer
vocabulary in `SamplingParams.update_from_tokenizer()` and raises a `bad_words` validation error if any token ID is out
of range.

 Without this check, an out-of-vocabulary bad word token ID can reach the sampler/logits masking path and fail later with
  a less clear engine-side error.

## Test Plan

```
cargo test -p vllm-text lower_sampling_params_rejects_out_of_vocab_bad_words
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.13s
     Running unittests src/lib.rs (target/debug/deps/vllm_text-b895ba6e49a8f640)

running 1 test
test lower::tests::lower_sampling_params_rejects_out_of_vocab_bad_words ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 58 filtered out; finished in 0.00s

```

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
@codex review

**Review by BugenZhao — APPROVED** [review]
The only source of `bad_words_token_ids` in the current implementation is `tokenize_bad_words`, so if there are any out-of-vocabulary tokens, it must be a bug in the tokenizer itself. But perhaps it is still useful to have this.
